eel/eel-string.c

```diff
@@ -325,7 +330,7 @@ eel_str_replace_substring (const char *string,
         memcpy (result_position, replacement, replacement_length);
```

Comment:
The nonnull warning here is misleading and slightly wrong: actually it is perfectly valid to pass `NULL` as source parameter (`replacement` here), but *only if* the length to copy is 0 (`replacement_length` here).  Which here is the case AFAICT.

This is probably a glitch of the way the analyzer finds this issue, thinking `dest` ought to be non-null.  If really needed, this could be worked around like `replacement == NULL ? "" : replacement`.

Reply:
I can't comment on new line 324 (old 319), so I do it here:

I don't get this warning at all, the only thing I could possibly imagine would be that the analyzer has a bug and fails to understand line 329 and things `result_position` gets assigned the value of `substring_position` or `p` somehow.  Otherwise it doesn't make sense as `result_position` is an offset in `result`, which is a newly allocated chunk of memory that cannot overlap any part of `string` (which `p` and `substring_position` are offsets of)